packages/ui-inputs/src/input.tsx

```diff
@@ -0,0 +1,221 @@
+            boxShadow: `0 0 0 1px ${theme.colors.cyan}`,
+        },
+    },
+    affixWrapper__focus: {
+        borderColor: `${theme.colors.cyan} !important`,
+        boxShadow: `0 0 0 1px ${theme.colors.cyan}`,
+    },
+    affixWrapper__disabled: {
+        boxShadow: 'none',
+        backgroundColor: theme.colors.disabled,
+        border: `1px solid ${theme.colors.greyscale5}`,
+    },
+    affixWrapper__error: {
+        borderColor: theme.colors.danger,
+        boxShadow: `0 0 0 1px ${theme.colors.danger}`,
+    },
+    preContent: {
+        color: theme.colors.greyscale3,
+        margin: '0 0 0 8px',
+        height: '24px',
+        minWidth: '24px',
+    },
+    postContent: {
+        margin: '0 8px 0 0',
+        flex: '1 0 auto',
+        height: '24px',
+        minWidth: '24px',
+        color: theme.colors.greyscale2,
+        fontSize: theme.input.fontSize,
+    },
+}), { theming });
+
+const renderAffix = (props: { element?: AffixContent, className?: string, supportBusy?: boolean }) => {
+    const { element, className, supportBusy } = props;
+    if (!element && !supportBusy) {
+        return null;
+    }
+    return <span className={className}>{element}</span>;
+};
+
+export const TextInput = (props: InputFieldProps) => {
+    const {
+        value,
+        defaultValue,
+        preContent,
+        postContent,
+        inputRef,
+        disabled,
+        isBusy,
+        supportBusy,
+        onChange,
+        placeholder,
+        hasError = false
+    } = props;
+    const theme = useTheme();
+    const classes = useStyles({ ...props, theme });
+    const hasAffix = !!preContent || !!postContent || supportBusy;
+    const canSetBusy = supportBusy && !disabled && !hasError;
+    
+    const [isInFocus, setIsInFocus] = useState(false);
+    
+    const inputClass = classNames({
+        [classes.input]: true,
+        [classes.inputDefault]: !hasError && !hasAffix,
+        [classes.inputError]: hasError && !hasAffix,
+        [classes.inputBorderLess]: hasAffix,
+    });
+    const affixWrapperClass = classNames({
+        [classes.affixWrapper]: true,
+        [classes.affixWrapper__focus]: !hasError && isInFocus,
+        [classes.affixWrapper__disabled]: disabled,
+        [classes.affixWrapper__error]: !disabled && hasError,
+    });
+    const prefixClass = classNames({ [classes.preContent]: true });
+    const suffixClass = classNames({ [classes.postContent]: true });
+    const hasErrorAttribute = hasError ? "true" : "false"; // convert to string for a custom attribute
+    const values = {}; // fix antd input value/defaultValue issue
+    if (value !== undefined) {
+        values['value'] = value;
+    }
+    if (defaultValue !== undefined) {
+        values['defaultValue'] = defaultValue;
+    }
+
+    const renderInput = () => <Input className={inputClass}
+                                     {...values}
+                                     defaultValue={defaultValue}
```

Comment:
this one is redudent as you handle `defaultValue` in `...values`

Reply:
a leftover, removed